pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -685,3 +696,36 @@ func getSpotVMOptions(spotVMOptions *v1beta1.SpotVMOptions) (compute.VirtualMach
 	// https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/spot-instances.md#eviction-policies
 	return compute.Spot, compute.Deallocate, billingProfile, nil
 }
+
+func (s *Reconciler) createAvailabilitySet() (string, error) {
+	// Try to find the availability set for the machine location
+	availabilityZones, err := s.availabilityZonesSvc.Get(context.Background(), &availabilityzones.Spec{
+		VMSize: s.scope.MachineConfig.VMSize,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	availabilityZonesSlice, ok := availabilityZones.([]string)
+	if !ok {
+		return "", fmt.Errorf("unexpected type %T", availabilityZones)
+	}
+
+	if len(availabilityZonesSlice) != 0 {
+		klog.V(4).Infof("No availability set needed for %s because availability zones were found", s.scope.Machine.Name)
+		return "", nil
+	}
+
+	klog.V(4).Infof("No availability zones were found for %s, an availability set will be created", s.scope.Machine.Name)
+
+	// Create the availability set
+	availabilityZoneName := fmt.Sprintf("%s-%s", s.scope.Machine.Labels[MachineSetLabelName], "worker")
```

Comment:
Why worker appended to the label? What does this result in? `worker` might already be in the name, perhaps we should append `availability-set`?

Reply:
I will leave machineset name only.